source/MRViewer/ImGuiHelpers.cpp

```diff
@@ -337,6 +338,148 @@ bool BeginStatePlugin( const char* label, bool* open, float width )
+    const float borderSize = style.WindowBorderSize * scaling;
+    const ImRect boundingBox( { pos.x + borderSize, pos.y + borderSize }, { pos.x + width - borderSize, pos.y + buttonSize - borderSize } );
+    
+    window->DrawList->PushClipRectFullScreen();
+    window->DrawList->AddRectFilled( boundingBox.Min, boundingBox.Max, bgColor );
+    window->DrawList->PopClipRect();
+
+    ImGui::PushStyleColor( ImGuiCol_Button, bgColor );
+    ImGui::PushStyleColor( ImGuiCol_Border, bgColor );   
+    ImGui::SetCursorPos( { MR::cDefaultItemSpacing * scaling * 2.0f/3.0f, 2.0f * scaling } );
+    ImGui::SetNextItemWidth( buttonSize );
+    ImGui::PushStyleVar( ImGuiStyleVar_ItemSpacing, { MR::cDefaultItemSpacing * scaling / 2.0f, MR::cDefaultItemSpacing * scaling } );
+    if ( ImGui::Button( *collapsed ? "\xef\x84\x85" : "\xef\x84\x87" ) ) // minimize/maximize button
+    {
+        *collapsed = !*collapsed;        
+    }
+    ImGui::SameLine();
+
+    if ( menu )
+    {
+        ImGui::PopFont();
+        titleFont = menu->getFontManager().getFontByType( MR::RibbonFontManager::FontType::SemiBold );
+        titleFont->Scale = 1.1f;
+        ImGui::PushFont( titleFont );
+    }
+
+    ImGui::SetCursorPosY( ImGui::GetCursorPosY() - style.FramePadding.y * 0.75f );
+    ImGui::Text( "%s", label);
+    ImGui::PopStyleVar();
+
+    if ( menu )
+    {
+        ImGui::PopFont();
+        titleFont->Scale = 1.0f;
+        ImGui::PushFont( iconsFont );
+    }
+    
+    ImGui::SameLine();    
+    ImGui::SetNextItemWidth( buttonSize );    
+    ImGui::SetCursorPosX( width - buttonSize );
+
+    ImGui::SetCursorPosY( ImGui::GetCursorPosY() + 1.0f * scaling );
+    if ( ImGui::Button( "\xef\x80\x8d" ) ) //close button
+    {
+        *open = false;
+        ImGui::PopFont();
+        ImGui::PopStyleColor( 2 );
+        ImGui::End();
+        return false;
+    }
+
+    if ( menu )
+    {
+        ImGui::PopFont();
+        iconsFont->Scale = 1.0f;
+    }
+
+    if ( *collapsed )
+    {
+        ImGui::PopStyleColor( 2 );
+        const auto borderColor = ImGui::ColorConvertFloat4ToU32( ImGui::GetStyleColorVec4( ImGuiCol_Border ) );
+
+        //ImGui doesn't draw bottom border if window is collapsed, so add it manually
+        window->DrawList->PushClipRectFullScreen();
+        window->DrawList->AddLine( { pos.x, pos.y + buttonSize - borderSize }, { pos.x + width, pos.y + buttonSize - borderSize }, borderColor, borderSize );
+        window->DrawList->PopClipRect();
+
+        ImGui::End();
+        return false;
+    }
+
+    ImGui::PopStyleColor( 2 );
+    
+    const ImGuiTableFlags tableFlags = ((height == 0.0f) ? ImGuiTableFlags_SizingStretchProp : ImGuiTableFlags_SizingStretchProp | ImGuiTableFlags_ScrollY );
+    const auto outerSize = ( height == 0.0f ) ? ImVec2 { 0, 0 } : ImVec2 { width - style.ScrollbarSize, height - style.WindowPadding.y };
+
+    ImGui::BeginTable( "ContentTable", 1, tableFlags, outerSize );
+    ImGui::TableNextColumn();
```

Comment:
do we need table now?

Reply:
yes, without table scrollbar will overlap titlebar